server/controllers/finance/vouchers.js

```diff
@@ -198,13 +242,85 @@ function report(req, res, next) {
  * @param {null|object} query The req.query object, retunrs list of voucher according the query,
  * @return promise
  */
-function getVouchers(uuid) {
+function getVouchers(uuid, request) {
   'use strict';
 
+  let detailed = request && !util.nullString(request.detailed) ? 1: 0;
+
+  // sql detailed or not for voucher
+  let sql = getSql(detailed);
+
+  // binary uuid
+  let bid = uuid ? db.bid(uuid) : null;
+
+  // sql params
+  let sqlParams = [];
+
+  if (request && (!util.nullString(request.dateFrom) && !util.nullString(request.dateTo)) && uuid) {
+
+    /**
+     * request is given but with dates, and uuid is also given :
+     * (request && (Boolean(request.dateFrom) && Boolean(request.dateTo)) && uuid)
+     */
+    sql += 'WHERE v.uuid = ? AND DATE(v.date) >= DATE(?) AND DATE(v.date) <= DATE(?)';
+    sqlParams = [];
+    sqlParams.push(bid);
+    sqlParams.push(request.dateFrom);
+    sqlParams.push(request.dateTo);
+
+  } else if ((request && (util.nullString(request.dateFrom) || util.nullString(request.dateTo)) && uuid) || (!request && uuid)) {
```

Comment:
Making local variables would help clear up this code:
```js
var hasRequest = !!request;
var hasDates =  (util.nullString(request.dateFrom) || util.nullString(request.dateTo);
var hasUuid = !!uuid;
```

Then you can do:
```js
if (hasRequest && hasDates && hasUuid) {
  // do something...
}
```
It would make your code much easier to read!